ci/purge_expired_sessions_from_datastore.yaml

```diff
@@ -0,0 +1,29 @@
+platform: linux
+image_resource:
+  type: docker-image
+  source:
+    repository: google/cloud-sdk
+params:
+  SERVICE_ACCOUNT_JSON: ((gcp.service_account_json))
+  PROJECT_ID:
+  DATAFLOW_TEMPLATE_VERSION:
+  EXPIRATION_TIME_OFFSET:
+run:
+  path: bash
+  args:
+    - -exc
+    - |
+      export GOOGLE_APPLICATION_CREDENTIALS=/root/gcloud-service-key.json
+      cat >$GOOGLE_APPLICATION_CREDENTIALS <<EOL
+      $SERVICE_ACCOUNT_JSON
+      EOL
+
+      gcloud auth activate-service-account --key-file "${GOOGLE_APPLICATION_CREDENTIALS}"
+      gcloud config set project "${PROJECT_ID}"
+      gcloud dataflow jobs run purgesession \
```

Comment:
I think this job should be blocking until the dataflow job has been completed then it should report the status back. At the moment, this would just run and without checking GCP console ourselves, we wouldn't know if it passed or failed. We should do what we do for backups and restore and report the state. The task should pass/fails based on the job state.

Reply:
Which is the blocking part in backups and restore in the script?

Reply:
The state of the job will go to 'JOB_STATE_PENDING' and at best takes 2 minutes to change to failed or succeeded, when census is running this could take 30+ minutes. I am not sure polling the job every few seconds to see the job state is a great idea

Reply:
The datastore `import` and `export` commands are synchronous by default (blocking), they do have an `--async` flag, but we chose not to use that, so we can report the progress. The dataflow by default is async so we won't know the state without polling. I am not sure if dataflow has a flag that will block it (worth checking). But even if we have to poll it, I don't see any issues with that, it would be the recommended way to automate it if it has no blocking option. We could easily just poll it every 60 seconds. 

I don't think it's very useful to have a CI job that triggers a job at 4am which will pass and report to Slack as passed but may have failed the actual job in the background. Without checking the state, someone will have to check the state manually every time to see if the job was actually successful.

Reply:
Dataflow doesn't have an export command, so I am going to have to call the describe every minute